pkg/services/pluginsintegration/pluginstore/store.go

```diff
@@ -11,7 +11,15 @@ import (
 	"github.com/grafana/grafana/pkg/plugins/manager/sources"
 )
 
-var _ plugins.Store = (*Service)(nil)
+var _ Store = (*Service)(nil)
+
+// Store is the publicly accessible storage for plugins.
+type Store interface {
+	// Plugin finds a plugin by its ID.
+	Plugin(ctx context.Context, pluginID string) (Plugin, bool)
+	// Plugins returns plugins by their requested type.
+	Plugins(ctx context.Context, pluginTypes ...plugins.Type) []Plugin
+}
```

Comment:
I would keep the Store interface in the plugins pkg, with the rest of interfaces since this is not a specific implementation (unless I am missing something)

Reply:
>The change pluginDTO -> plugin sounds good to me.

🎉 

>The change of location of the plugin definition, I am not 100% sure because it's referenced in so many places that I don't know if it would qualify as "grafana OSS exclusive" but it may be the case so I am fine with that.

Yeah it basically it is only used by other parts of Grafana that wants to know info about a plugin (read-only).

>The Store interface change of location is the one that feels more weird, if it's an interface (and it's referenced by other Grafana components), why move it to pluginsintegrations?


The store was created (along with the DTO) so as to not leak internal plugins information outside to other parts of the Grafana codebase. The store and DTO are actually tightly coupled together, and when I thought about leaving one, it lead to package tangling. In reality, it makes sense to keep them together - otherwise, if we keep the interface in `pkg/plugins`, then it will have a dependency on `pluginsintegration` for the DTO. 

> if it's an interface (and it's referenced by other Grafana components), why move it to pluginsintegrations?

I think that's the key - it's only referenced by other Grafana components, it's not a core piece of the plugins functionality. For most consumers of the library, the registry is what you probably want. If you need to wrap that for whatever reason, then you can do so if you want to. 

The only reason the store has a corresponding interface is just for all the good reasons we use interfaces within Grafana (testing, portability, etc.), but at this point it could just as easy be the concrete implementation!

Reply:
okay, I understand why you did it due to the dependency `store` -> `Plugin`. Feels a bit in the limit of what should be in `pluginsintegrations` (at least from my understanding) but feel free to go